### ISSUE TITLE
Stop download after cloud execution

### ISSUE THREAD

**Issue by biotechjs** [snakemake/snakemake#257]
When running on the cloud, but I seem to have instructed snakemake to download the results from the cloud at the end of the execution. Is it possible to simply have the output in `default-remote-prefix` and not downloading them?

Here an example:
`snakemake --tibanna --default-remote-prefix ${BUCKET}/try --jobs 20  --default-remote-provider S3 --no-shared-fs `

with this a small `Snakefile`:

```
samples = list(range(2))
  
rule all:
    input:
       expand("output/{sample}.out", sample=samples)

rule hostname:
    output:
        "output/{sample}.out"
    shell:
        "hostname > {output}"
```

**Comment by dcroote** [comment]
Downloading remote files happens because `rule all` is not being executed in the cloud, but rather [locally](https://snakemake.readthedocs.io/en/master/snakefiles/rules.html?highlight=localrules#local-rules) as `localrule all` (this can be seen in the snakemake run log). 

The following `rule all` will execute remotely as it has a `shell` directive, thus preventing local file download. Note the `output` placeholder file is to prevent the job from being force-run upon each invocation:

```python
rule all:
    input:
        expand("output/{sample}.out", sample=samples)
    output:
        "output/workflow-complete.txt"
    shell:
        "touch {output}"
```

This isn't a great solution as it just moves file download to the remote job rather than avoiding it; ideally when using `--default-remote-prefix` snakemake would detect that remote input files are present and, given no `shell` directive or `output`, provide the typical response `Nothing to be done.` Thoughts @johanneskoester ?

**Comment by peter-fsdx** [comment]
I am coming up against this problem now. I have an large workflow that generates a lot of data. After running efficiently in parallel on the cluster the rule all runs forever as it downloads all of the files from remote. It's not the end of the world but certainly annoying.

Would love an option to prevent downloading the inputs of the first tule

**Comment by m-jahn** [comment]
Tibanna in Snakemake is superseded by the use of the [AWS executor plugins](https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/aws-batch.html). I thinks this can be closed.
